types/config/index.d.ts

```diff
@@ -16,7 +16,7 @@ declare namespace c {
     // see https://github.com/lorenwest/node-config/wiki/Using-Config-Utilities
     interface IUtil {
         // Extend an object (and any object it contains) with one or more objects (and objects contained in them).
-        extendDeep(mergeInto: any, mergeFrom: any, depth?: number): any;
+        extendDeep(mergeInto: any, ...mergeFrom: any): any;
```

Comment:
What about defining multiple declarations to support the depth parameters for some numbers of arguments. Eg.
```
extendDeep(mergeInto: any, mergeFrom: any, depth?: number): any;
extendDeep(mergeInto: any, mergeFrom1: any, mergeFrom2: any, depth?: number): any;
extendDeep(mergeInto: any, ...mergeFrom: any): any;
```

Reply:
This suggestion works in my use case and passes the tests here. While not technically correct, it covers more of the typical use cases and hints at the true interface which the user can confirm in the documentation. I'll update the PR with the suggestion. Thanks.